### PR TITLE
[INFRA-2619][WIP] Release component jenkinsci/remoting

### PR DESCRIPTION
**Pull request by olblak**
This pull request makes it a little bit more generic in order to use scripts in this repository for releasing jenkins components as well.
I was able to test locally all the parts including publishing artifacts on repo.jenkins-ci.org in a private repository.

The only part that I wasn't able to easily reuse are `utils/release.sh --verifyGPGSignature` and `utils/release.sh --verifyCertificateSignature` as we need to know in advance artifact name and at the moment they contain version numbers like `remoting-4.4.jar` vs `remoting.jar` 

In order for this pull request to, we also have to configure maven release plugin on jenkinsci/remoting  as proposed in this [PR](https://github.com/jenkinsci/remoting/pull/381) and obviously also configure release.ci.jenkins.io  

https://issues.jenkins-ci.org/browse/INFRA-2619

### CODE REVIEW

File: Jenkinsfile.d/components/remoting [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            }
+
+          }
+        }
+      }
+      stage('Stage Release') {
+        steps {
+          container('maven') {
+            sh '''
+              utils/release.sh --stageRelease
```
Comment (timja) [inline Jenkinsfile.d/components/remoting:107]: any reason this is staged and not just done in one job?

Comment (olblak) [reply Jenkinsfile.d/components/remoting:107]: It's just to reuse existing script, `--stageRelease` which does run release:stage, I could easily add a new parameter named `--performRelease` to run mvn release:perform then I wouldn't need https://github.com/jenkinsci/remoting/pull/381.

Comment (timja) [reply Jenkinsfile.d/components/remoting:107]: Why not do that? I can't see why you wouldn't want to just perform the release?

Comment (olblak) [reply Jenkinsfile.d/components/remoting:107]: It won't take too much time to add this

**Comment by timja** [comment]
Looks good, anything left on it?

**Comment by olblak** [comment]
> Looks good, anything left on it?

Nope

**Pull request merged**